draft-ietf-webpush-protocol.xml

```diff
@@ -1270,16 +1261,23 @@ HEADERS      [stream 82] +END_STREAM
 
      <section title="Denial of Service Considerations">
        <t>
-         Discarding unwanted messages at the user agent based on message
-         authentication doesn't protect against a denial of service attack on
-         the user agent.  Even a relatively small volume of push messages can
-         cause battery-powered devices to exhaust power reserves.
+         End-to-end confidentiality mechanisms, such as those in <xref
+         target="API"/>, prevent an entity with a valid push message
+         subscription URI from learning the contents of push messages.  Push
+         messages that are not successfully authenticated will not be delivered
+         by the API, but this can present a denial of service risk. Even a
+         relatively small volume of push messages can cause battery-powered devices
+         to exhaust power reserves.
        </t>
        <t>
-         An application can limit where valid push messages can originate by
-         limiting the distribution of push URIs to authorized entities.
+         A user agent can control where valid push messages originate by
+         limiting the distribution of push URIs to authorized application servers.
          Ensuring that push URIs are hard to guess ensures that only application
-         servers that have been given a push URI can use it.
+         servers that have received a push URI can use it. A further refinement is
+         <xref target="I-D.ietf-webpush-vapid">Voluntary Application Server Identification</xref>,
+         which allows a user agent to restrict a subscription to a specific application
+         server. The push service then rejects any requests for message delivery with missing
+         or invalid credentials.

```

Comment:
Good point. This ties in nicely with the issue of **when** a message is discarded.